habitat-lab/habitat/tasks/rearrange/actions/actions.py

```diff
@@ -417,8 +417,9 @@ def update_base(self):
             rigid_state.translation, target_rigid_state.translation
         )
 
-        # Offset the base
-        end_pos -= self.cur_articulated_agent.params.base_offset
+        # Offset the base if the base height is different between end_pos and the current state
+        if end_pos[1] != rigid_state.translation[1]:
```

Comment:
Have you tested this in a scene with different floor heights? At a glance, it's not really intuitive why this fix is necessary or correct.